README.md

```diff
@@ -1341,6 +1342,78 @@ app.UseSwaggerUI(c =>
 });
 ```
 
+### Change Swagger UI Config ###
+
+Swagger UI has a big variety of options and a mighty plugin API allowing customizations. 
+To customize the Swagger UI (and OAuth) configuration objects before they are passed into the Swagger UI initialization,
+inject a custom JavaScript file which defines a global function with the following signature.
+In many cases this feature allows to avoid using a custom `index.html` in your project.
+
+```js
+/**
+ * @param configObject See https://swagger.io/docs/open-source-tools/swagger-ui/usage/configuration/
+ * @param oauthConfigObject See https://swagger.io/docs/open-source-tools/swagger-ui/usage/oauth2/
+ */
+function initConfig(configObject, oauthConfigObject) { }
+```
+
+Through this mechanism any options that Swagger UI supports can be set and also plugins can be registered. 
+Swagger UI is written with React so it can be a bit tricky to write custom components. 
+Here a full example which adds a custom section to the parameters of all operations:
+
+```js
+// swagger-ui.js
+// https://swagger.io/docs/open-source-tools/swagger-ui/customization/plugin-api/
+function configInit(config) {
+    if(!config.plugins) {
+        config.plugins = [];
+    }
+    config.plugins.push(TopbarPlugin);
+}
+
+const TopbarPlugin = (system) => {
+    return {
+        // Register an own React component 
+        components: {
+            CustomOperationDetails: class CustomOperationDetails extends system.React.Component {
+                render() {
+                    const hello = this.props?.operation?.get('x-hello');
+                    if(!hello) {
+                        return '';
+                    }
+                    return system.React.createElement('div',
+                        { className: 'hello' },
+                        hello
+                    );
+                }
+            }
+        },
+        wrapComponents: {
+            // wrap the parameters component
+            parameters: Original => (props => {
+                // get the custom React component
+                const CustomOperationDetails = props.getComponent('CustomOperationDetails');
+                // Use the non JSX React API (https://reactjs.org/docs/react-without-jsx.html)
+                // to create a wrapper div holding our own component and the original parameters component
+                return system.React.createElement('div',
+                    { className: "parameters-wrap"},
+                    system.React.createElement(CustomOperationDetails, props),
+                    system.React.createElement(Original, props)
+                );
```

Comment:
This example looks out-of-date compared to the example in the non-obsolete React documentation.